Error formatting hangs indefinitely with `regex: true`

I'm running into case where the plugin seems to hang indefinitely. I did a little digging and inspected with `breakpoint()`.

This case reproduces the forever loop, I didn't manage to reduce this further, it seems it depends somewhat on producing notes with asterisks (`*`) in them and I'm not entirely sure of what the logic within mypy is for that (it seems to vary when `disable_cache: true` isn't given ...).

```yaml
- case: repro
  disable_cache: true
  regex: true
  main: |
    from predicates import is_valid_name
    reveal_type(is_valid_name)
    is_valid_name(1)
  out: |
    main:2: note: Revealed type is "def \(builtins\.str.?\) -> builtins\.bool.?"
    main:3: error: Argument 1 has incompatible type "int"; expected "str"  [arg-type]
  files:
    - path: compose.py
      content: |
        from typing import TypeVar, Callable
        AA = TypeVar("AA")
        AR = TypeVar("AR")
        BA = TypeVar("BA")
        def compose2(a: Callable[[AA], AR], b: Callable[[BA], AA]) -> Callable[[BA], AR]:
            ...
    - path: predicates.py
      content: |
        from typing import Callable
        from compose import compose2
        Predicate = Callable[[object], bool]
        as_parts: Callable[[str], list[str]] = ...
        is_valid_parts: Predicate[list[str]] = ...
        is_valid_name = compose2(is_valid_parts, as_parts)
```

This is the loop that keeps running forever, because it's given two identical strings: https://github.com/typeddjango/pytest-mypy-plugins/blob/337d8048fe02cbf68df7de69337999ee61af05d3/pytest_mypy_plugins/utils.py#L148

I think this might not be a high-priority issue, because I think the pattern might not make sense, nevertheless, it seems sensible to have a guard in place at some level.

--- a/pytest_mypy_plugins/tests/test-regex_assertions.yml
+++ b/pytest_mypy_plugins/tests/test-regex_assertions.yml
@@ -15,6 +15,20 @@
   out: |
     main:2: note: .*str.*
 
+- case: regex_with_out_does_not_hang
+  expect_fail: yes
+  regex: yes
+  main: |
+    'abc'.split(4)
+  out: |
+    main:1: error: Argument 1 to "split" of "str" has incompatible type "int"; expected "Optional[str]"
+
+- case: regex_with_comment_does_not_hang
+  expect_fail: yes
+  regex: yes
+  main: |
+    a = 'abc'.split(4)  # E: Argument 1 to "split" of "str" has incompatible type "int"; expected "Optional[str]"
+
 - case: expected_single_message_regex
   regex: no
   main: |
@@ -34,7 +48,7 @@
     a = 'hello'
     reveal_type(a)  # N: .*str.*
 
-- case: regext_does_not_match
+- case: regex_does_not_match
   expect_fail: yes
   regex: no
   main: |

--- a/pytest_mypy_plugins/utils.py
+++ b/pytest_mypy_plugins/utils.py
@@ -144,6 +144,8 @@ def _add_aligned_message(s1: str, s2: str, error_message: str) -> str:
 
     error_message += "Alignment of first line difference:\n"
 
+    assert s1 != s2
+
     trunc = False
     while s1[:30] == s2[:30]:
         s1 = s1[10:]
@@ -260,7 +262,9 @@ def assert_expected_matched_actual(expected: List[OutputMatcher], actual: List[s
             format_error_lines(actual_message_lines), format_error_lines(expected_message_lines)
         )
 
-        if (
+        if expected_line and expected_line.regex:
+            error_message += "The actual output does not match the expected regex."
+        elif (
             first_diff_actual is not None
             and first_diff_expected is not None
             and (

